test/unit/test_checkpointing.py

```diff
@@ -172,6 +172,28 @@ def tearDown(self):
         shutil.rmtree(self.test_dir, ignore_errors=True)
 
 
+class TestSaveIntermediateModel(unittest.TestCase):
```

Comment:
Yes, they were removed because I changed how we were saving models for this class(pickling vs save_model). There are noted problems with pickling mock objects for testing, which we were doing now with this change. I tried quite a few fixes for it, but wasn't able to get it to work and removed it. This is being tested in the functional tests(CR-28179853).

I kept this here to make sure we were testing the setting up of the callback at least, but since this is being tested in functional tests we can remove it from unit tests.

Reply:
The problem is our deployment pipeline currently does not run container tests in our QA package. CodeBuild runs unit + container tests in this repository, and deployment pipeline only runs integration tests. If there no integration tests, container tests should be added [here](https://github.com/aws/sagemaker-xgboost-container/tree/master/test/integration/local) in this repository so we have automated tests. Also, maybe I'm missing something, but based on your comment on `HyperparameterTuner. early_stopping_type='AUTO'` sending a SIGTERM, I don't see why we can't have integration tests.